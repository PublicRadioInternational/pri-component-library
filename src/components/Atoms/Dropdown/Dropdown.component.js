/**
 * @file Dropdown.component.js
 * Creates dropdown component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Dropdown.css';
import Button from '../Button/Button.component';

// const Dropdown = ({ items, btnTitle, isToggled }) => (
class Dropdown extends React.Component {
  constructor(props) {
    super(props);
    this.handleToggleChange = this.handleToggleChange.bind(this);
    this.state = { toggled: false };
  }
  handleToggleChange() {
    this.setState(prevState => ({
      toggled: !prevState.toggled
    }));
  }
  render() {
    return (
      <div className={styles.btnGroup}>
        <Button
          hidden={false}
          title={this.props.btnTitle}
          url="#"
          className="btnGrpOrange"
          ariaHaspopup
          ariaExpanded={false}
        />
        <Button
          button
          hidden
          title="Toggle Dropdown"
          url="#"
          className="btnDropdownOrange"
          ariaHaspopup
          ariaExpanded={false}
          toggled={this.handleToggleChange}
        />
        <div
          className={styles.dropdown}
          data-toggled={this.state.toggled ? 'toggled' : null}
        >
          {this.props.items.map(item => (
            <a className={styles.dropdownItem} href={item.url} key={item.id}>
              {item.title}
            </a>
          ))}
        </div>
      </div>
    );
  }
}

Dropdown.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  btnTitle: PropTypes.string.isRequired
};

export default Dropdown;
