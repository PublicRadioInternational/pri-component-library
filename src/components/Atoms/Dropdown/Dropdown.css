/* import colors... */
@value colors: "../../00_global/colors.css";
@value blue from colors;
/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value copy from fonts;

/* Group */
.btnGroup {
  display: flex;
  position: relative;
  vertical-align: middle;
}

/* Dropdown */
.dropdown {
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid rgba(0, 0, 0, 0.15);
  border-radius: 0.25rem;
  display: none;
  left: 0;
  margin: 0.125rem 0 0;
  min-width: 10rem;
  padding: 0.5rem 0;
  position: absolute;
  top: 100%;
  z-index: 1;
}

.dropdownItem {
  clear: both;
  color: blue;
  display: block;
  font-size: 0.875em;
  font-weight: 400;
  padding: 0.25rem 1.5rem;
  width: 100%;
}
