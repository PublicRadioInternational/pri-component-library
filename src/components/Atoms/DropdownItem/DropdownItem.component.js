/**
 * @file DropdownItem.test.js
 * Exports a dropdown item component.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './DropdownItem.css';

/**
 * Component that renders a dropdown menu item.
 */
export default class DropdownItem extends Component {
  static propTypes = {
    children: PropTypes.node,
    url: PropTypes.string,
    heading: PropTypes.string,
    onClick: PropTypes.func
  };

  static defaultProps = {
    children: [],
    url: null,
    heading: null,
    onClick: () => {}
  };

  render() {
    const { children, url, onClick, heading } = this.props;
    if (url) {
      return (
        <a className={styles.dropdownItem} href={url} onClick={onClick}>
          {children}
        </a>
      );
    } else if (heading) {
      return <h3 className={styles.dropdownHeading}>{heading}</h3>;
    }
    return <hr className={styles.dropdownHr} />;
  }
}
