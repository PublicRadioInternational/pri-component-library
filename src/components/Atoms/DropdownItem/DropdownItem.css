/* import colors... */
@value colors: "../../00_global/colors.css";
@value blue, orange, white, grayLighter, blueLight, grayDark from colors;

/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value heading from fonts;

.dropdownItem,
.dropdownHeading,
.dropdownSpace {
  padding: 0.4rem 1.5rem;
}

.dropdownItem {
  clear: both;
  color: blue;
  display: block;
  font: 400 0.875em heading;
  width: 77%;
  border: none;
  cursor: pointer;
  text-decoration: none;
}

.dropdownItem:hover {
  background-color: blueLight;
  color: white;
}

.dropdownHeading {
  color: grayDark;
  font: 800 0.875em heading;
  font-weight: bold;
  margin: 0;
}

.dropdownHr {
  border: none;
  height: 1px;
  background-color: grayLighter;
}

.dropdownForm {
  width: 225px;
}
