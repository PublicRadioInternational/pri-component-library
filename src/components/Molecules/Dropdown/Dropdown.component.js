/**
 * @file Dropdown.component.js
 * Exports a dropdown component.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Downshift from 'downshift';
import styles from './Dropdown.css';
import Button from '../../Atoms/Button/Button.component';

/**
 * Component that renders a Dropdown menu button.
 */
export default class Dropdown extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    children: PropTypes.node,
    onClick: PropTypes.func,
    url: PropTypes.string,
    color: PropTypes.oneOf(['Orange', 'White']),
    icon: PropTypes.string
  };

  static defaultProps = {
    children: [],
    color: 'White',
    url: null,
    onClick: () => {},
    icon: null
  };

  render() {
    const { title, onClick, children, color, url, icon } = this.props;

    return (
      <Downshift>
        {({ isOpen, getButtonProps }) => (
          <div className={styles.dropdownGrp}>
            <Button
              className={styles.btnGrp}
              url={url}
              color={color}
              onClick={onClick}
              icon={icon}
            >
              {title}
            </Button>
            <Button
              {...getButtonProps()}
              className={styles[`btnDropdown${color}`]}
              color={color}
            />
            {isOpen ? <div className={styles.dropdown}>{children}</div> : null}
          </div>
        )}
      </Downshift>
    );
  }
}
