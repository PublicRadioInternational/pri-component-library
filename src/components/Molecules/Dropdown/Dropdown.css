/* import colors... */
@value colors: "../../00_global/colors.css";
@value blue, orangeDark from colors;

/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value copy from fonts;

/* import breakpoints */
@value header as bp-header from "../../00_global/breakpoints.css";

/* Group */
.dropdownGrpSmall {
  display: none;
}

.dropdownGrp {
  display: flex;
  justify-content: flex-end;
  position: relative;
  vertical-align: middle;
}

/* Dropdown */
.dropdown {
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid rgba(0, 0, 0, 0.15);
  border-radius: 0.25rem;
  margin: 0.125rem 0 0;
  min-width: 13rem;
  padding: 0.5rem 0;
  position: absolute;
  right: 0;
  top: 100%;
  z-index: 1;
}

@media bp-header {
  .btnGrp {
    border-radius: 0.25rem 0 0 0.25rem;
  }
}

.btnDropdown {
  border-radius: 0 0.25rem 0.25rem 0;
  margin-left: -1px;
  padding: 0.3rem 0.6rem;
}

.btnDropdown::after {
  content: '';
  display: block;
  width: 0;
  height: 0;
  border-top: 0.4em solid blue;
  border-right: 0.3em solid transparent;
  border-bottom: 0;
  border-left: 0.3em solid transparent;
}

.btnDropdownWhite {
  composes: btnDropdown;
}

.btnDropdownOrange {
  composes: btnDropdown;
  border-left: 1px solid orangeDark;
}

.btnDropdownOrange::after {
  border-top: 0.4em solid white;
}

.btnHide {
  display: none;
}

@media bp-header {
  .btnHide {
    display: inline-block;
  }
}
