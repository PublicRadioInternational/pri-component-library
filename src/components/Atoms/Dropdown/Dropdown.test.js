/**
 * @file Dropdown.test.js
 * Contains tests for Dropdown.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown.component';
import DropdownItem from '../DropdownItem/DropdownItem.component';

describe('<Dropdown />', () => {
  const dropdown = () => (
    <Dropdown url="https://google.com" title="Listen" onClick={() => {}}>
      <DropdownItem title="Google" url="https://google.com" />
      <DropdownItem title="Custom Action" onClick={() => {}} />
    </Dropdown>
  );

  it('Matches the snapshot', () => {
    const component = renderer.create(dropdown());
    expect(component.toJSON()).toMatchSnapshot();
  });
});
