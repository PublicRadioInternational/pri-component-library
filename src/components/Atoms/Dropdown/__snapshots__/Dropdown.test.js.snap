// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Matches the snapshot 1`] = `
<div
  className="dropdownGrp"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="btnGrpWhite"
    onClick={[Function]}
    type="button"
  >
    <span
      className={null}
    >
      Listen
    </span>
  </button>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="btnDropdownWhite"
    onClick={[Function]}
    type="button"
  >
    <span
      className="hidden"
    >
      Toggle Listen Dropdown
    </span>
  </button>
  <div
    className="dropdown"
    data-open={false}
  >
    <a
      className="dropdownItem"
      href="https://google.com"
      onClick={[Function]}
    >
      Google
    </a>
    <a
      className="dropdownItem"
      href={null}
      onClick={[Function]}
    >
      Custom Action
    </a>
  </div>
</div>
`;
